cluster-autoscaler/processors/nodes/scale_down_node_processor.go

```diff
@@ -0,0 +1,20 @@
+package nodes
+
+import (
+	apiv1 "k8s.io/api/core/v1"
+
+	"k8s.io/autoscaler/cluster-autoscaler/context"
+	"k8s.io/autoscaler/cluster-autoscaler/utils/errors"
+)
+
+type ScaleDownNodeProcessor interface {
+	// GetHarborCandidates returns nodes that potentially could harbor the pods that would become
+	// unscheduled after a scale down.
+	GetHarborCandidates(*context.AutoscalingContext, []*apiv1.Node) ([]*apiv1.Node, errors.AutoscalerError)
```

Comment:
I don't like this name too much - it's not intuitive at all to me what this does. I'd prefer something like GetPodDestinationCandidates(). 
It may be just me though - let's get another opinion. 
@aleksandra-malinowska @krzysztof-jastrzebski - WDYT?

Reply:
Not great with names. Also, I don't know if there can be a very intuitive name for the concept. Changed to GetPodDestinationCandidates for now